src/language-server/definition-finder.ts

```diff
@@ -103,41 +124,55 @@ export default class DefinitionFinder extends Handler {
    */
   private async getReferences(params: ReferenceParams): Promise<Location[]> {
     const locations: Location[] = [];
-    if (params.context.includeDeclaration) {
-      const definition = await this.getDefinition(params);
-      if (definition) {
-        if (Array.isArray(definition)) {
-          locations.push(...definition);
-        } else {
-          locations.push(definition);
-        }
-      }
-    }
     const localPath =
         this.converter.getWorkspacePathToFile(params.textDocument);
     const position = this.converter.convertPosition(params.position);
     const analysis = await this.analyzer.analyzePackage();
+    if (params.context.includeDeclaration) {
+      locations.push(...await this.getDefinition(params, analysis));
+    }
+
     const document = analysis.getDocument(localPath);
     if (!(document instanceof Document)) {
       return locations;
     }
-    const location =
+    const astResult =
         await this.featureFinder.getAstAtPosition(document, position);
-    if (!location) {
+    if (!astResult) {
       return locations;
     }
-    if (location.kind === 'tagName') {
-      const ranges = [...analysis.getFeatures({
-                       kind: 'element-reference',
-                       id: location.element.tagName!,
-                       externalPackages: true,
-                     })].map(e => e.sourceRange);
-      locations.push(...ranges.map(r => {
-        return {
-          uri: this.converter.getUriForLocalPath(r.file),
-          range: this.converter.convertPRangeToL(r)
-        };
-      }));
+    if (astResult.language === 'css') {
+      const featureResult =
+          this.featureFinder.getFeatureForAstLocation(astResult, position);
+      if (featureResult) {
+        if (featureResult.feature instanceof CssCustomPropertyAssignment ||
+            featureResult.feature instanceof CssCustomPropertyUse) {
+          const propertyUses = [...analysis.getFeatures({
+            kind: 'css-custom-property-use',
+            id: featureResult.feature.name
+          })];
+          locations.push(...propertyUses.map(f => {
+            return {
+              uri: this.converter.getUriForLocalPath(f.sourceRange.file),
+              range: this.converter.convertPRangeToL(f.sourceRange)
+            };
+          }));
+        }
+      }
+    } else {
+      if (astResult.node.kind === 'tagName') {
```

Comment:
nit: could be merged with the line above into a single `else if (...`

Reply:
+1